client/src/app/pages/applications/assessment-actions/components/dynamic-assessment-button.tsx

```diff
@@ -0,0 +1,134 @@
+import { Paths, formatPath } from "@app/Paths";
+import { Application, InitialAssessment, Questionnaire } from "@app/api/models";
+import {
+  useCreateAssessmentMutation,
+  useFetchAssessmentsByAppId,
+} from "@app/queries/assessments";
+import { Button } from "@patternfly/react-core";
+import React, { FunctionComponent } from "react";
+import { useHistory } from "react-router-dom";
+import spacing from "@patternfly/react-styles/css/utilities/Spacing/spacing";
+
+enum AssessmentAction {
+  Take = "Take",
+  Retake = "Retake",
+  Continue = "Continue",
+}
+
+interface DynamicAssessmentButtonProps {
+  questionnaire: Questionnaire;
+  application: Application;
+}
+
+const DynamicAssessmentButton: FunctionComponent<
+  DynamicAssessmentButtonProps
+> = ({ questionnaire, application }) => {
+  const history = useHistory();
+  const { assessments } = useFetchAssessmentsByAppId(application?.id);
+  const matchingAssessment = assessments?.find(
+    (assessment) => assessment.questionnaire.id === questionnaire.id
+  );
+
+  const onSuccessHandler = () => {};
+  const onErrorHandler = () => {};
+
+  const { mutateAsync: createAssessmentAsync } = useCreateAssessmentMutation(
+    onSuccessHandler,
+    onErrorHandler
+  );
+
+  const determineAction = () => {
+    if (!application) {
+      console.error("Application is undefined. Cannot proceed.");
+      return AssessmentAction.Take;
+    }
+
+    if (!matchingAssessment) {
+      return AssessmentAction.Take;
+    } else if (matchingAssessment.status === "STARTED") {
+      return AssessmentAction.Continue;
+    } else {
+      return AssessmentAction.Retake;
+    }
+  };
+
+  const determineButtonBackgroundColor = () => {
+    const action = determineAction();
+    if (action === AssessmentAction.Continue) {
+      return "#5ba353"; // Continue background color
+    } else if (action === AssessmentAction.Retake) {
+      return "#F0AB01"; // Retake background color
+    } else {
+      return ""; // Default background color for "Take"
+    }
+  };
```

Comment:
I wonder if instead we should conditionally apply CSS classNames to the buttons and define these colors in CSS?

Also we could use the CSS variables from the PF color palette instead of hard coded hash values. That way if we ever introduce dark mode the colors should be able to adjust automatically.